### PR TITLE
Return found nodes

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Return the nodes found by a `findNodes` or `searchForNewPeers` request so they can be utilised even if they didn't get added to the node table (eg because it was full).

### CODE REVIEW

File: src/main/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandler.java [hunk 1]
```diff
@@ -3,30 +3,112 @@
...
+    return existingRecord.isEmpty()
+        || existingRecord.get().getNode().getSeq().compareTo(newRecord.getSeq()) < 0;
+  }
+
+  private boolean isValid(final NodeRecord record) {
+    if (!record.isValid()) {
+      logger.debug("Rejecting invalid node record {}", record);
+      return false;
+    }
+    return true;
```
Comment (rolfyone) [inline src/main/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandler.java:99]: nit: I'd return `record.isValid()`, and use the condition just as a decision to debug...

Comment (rolfyone) [reply src/main/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandler.java:99]: although that admittedly does introduce a chance to have a change in value, maybe that's why this was done this way..

File: src/test/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    assertThat(handler.handleResponseMessage(message, session)).isTrue();
+
+    final NodeRecordInfo nodeRecordInfo = NodeRecordInfo.createDefault(updatedRecord);
+    verify(nodeTable).save(nodeRecordInfo);
+    verify(session, never()).updateNodeRecord(any());
+    verify(session, never()).putRecordInBucket(any());
+  }
+
+  @Test
+  public void shouldUpdateNodeRecordInSessionWhenSelfEnrReturnedWithHigherSeqNum() {
```
Comment (rolfyone) [inline src/test/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandlerTest.java:79]: `whenOwnEnrReturned`...

**Review by rolfyone — APPROVED** [review]
LGTM

File: src/main/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandler.java (hunk 1 above)
Comment (ajsutton) [reply src/main/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandler.java:99]: Calling `isValid` requires performing potentially expensive cryptographic operations, hence the single call.  Potentially validity could be cached but it currently isn't.

File: src/test/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandlerTest.java (hunk 2 above)
Comment (ajsutton) [reply src/test/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandlerTest.java:79]: I meant to just rename this but forgot to actually push the change.  Self ENR sounds weird at times but it is actually the term used for the node's local ENR in most places.

File: src/main/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandler.java (hunk 1 above)
Comment (rolfyone) [reply src/main/java/org/ethereum/beacon/discovery/pipeline/info/FindNodeResponseHandler.java:99]: true, but we can still call once by 
```
boolean result = record.isValid();
if (! result) ...

return isValid;
```

**Pull request merged**